### PR TITLE
Fix #6242: Scenery invalidated when there is no visual change

### LINKED ISSUE #6242: Grass and scenery invalidated when there is no visual change
Each frame, various tiles are updated to increase the grass length or to wilt the flowers. When this happens the tile is invalidated, however there isn't always a change in appearance.

To improve performance, it should only invalidate the tile if necessary. You can see how often it invalidates the tiles by turning on dirty drawing visuals via the debug paint window.

image

### PR DESCRIPTION
**Pull request by rwjuk**
This PR skips invalidating tiles when grass 'grows' where there is no visual change, and also likewise skips invalidation for scenery aging except when the 'withering' thresholds are crossed. This provides a minor performance boost generally, and a pretty decent one under pathological circumstances (loads of scenery, particularly flowers, and high time acceleration).

Strictly speaking only 'witherable' scenery should be invalidated as it ages, but checking for that flag is too expensive and cancels out most of the performance improvement!

### CODE REVIEW

File: src/openrct2/world/map.c [hunk 1]
```diff
@@ -3495,9 +3495,26 @@ static void map_update_grass_length(sint32 x, sint32 y, rct_tile_element *tileEl
...
+    
+    // If the new grass length won't result in an actual visual change
+    // then skip invalidating the tile, no point
+    if (((oldLength > 0 && oldLength < 4) && (newLength > 0 && newLength < 4)) ||
+        ((oldLength > 3 && oldLength < 7) && (newLength > 3 && newLength < 7)))
+    {
+        return;
+    }
+
+    sint32 z0, z1;
```
Comment (AaronVanGeffen) [inline src/openrct2/world/map.c]: Could you initialise these directly to the values right below?

Comment (rwjuk) [reply src/openrct2/world/map.c]: Could do better than that - we don't even need two vars there really...

**Comment by AaronVanGeffen** [comment]
Thanks for taking the time to look into this! Not invalidating sprites when there is no observable change seems obvious in hindsight, but who would have thought it would have such a noticeable impact! 😃

**Comment by Gymnasiast** [comment]
Could you update the changelog as well? I'd put it under improved and mention it slightly improves performance. Oh, and can you include an entry for SSE 4.1 as well?

**Comment by rwjuk** [comment]
@Gymnasiast Done. (I cancelled the CI runs as I forgot the `[ci skip]`)

**Comment by duncanspumpkin** [comment]
I'm surprised that getting a scenery entry would be more expensive than a redraw of an area.

**Comment by rwjuk** [comment]
@duncanspumpkin I'll retest - the trouble with these sort of tweaks is that the differences are so small anyway that it's easy to get lost in the noise!

**Comment by janisozaur** [comment]
@rwjuk that sounds like an issue with inadequate testing. I really wish there was some more benchmarking. `benchgfx` only gets us so far.

**Comment by rwjuk** [comment]
@duncanspumpkin Tweaked conditional for the entry check added.

**Comment by Gymnasiast** [comment]
@rwjuk You can merge this when you're done, AFAIC.

**Comment by rwjuk** [comment]
@Gymnasiast Just waiting for CI and then I will :)

**Pull request merged**